File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curr_sum = nums[0]
        max_sum = nums[0]
        
        for num in nums[1:]:
            curr_sum = max(curr_sum + num, num)
            max_sum = max(max_sum, curr_sum)
            
        return max_sum
        

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int curr_sum = nums[0];
        int max_sum = nums[0];
        int n = nums.size();
        
        for(int i=1; i<n; i++){
            curr_sum = max(curr_sum+nums[i], nums[i]);
            max_sum = max(curr_sum, max_sum);
        }
        
        return max_sum;
    }
};